### DISCUSSION TITLE
Downloads through the "Download tarball" button are unresponsive

### DISCUSSION THREAD

**Discussion by jeremydiazmeta** [verdaccio/verdaccio#4068] [Q&A] [answered]
Hello,

We are running into issues when users try to download packages through the 'Download tarball' buttons. Whenever someone clicks the button the download happens in the background and there is no indication of the package being downloaded until the download finishes. With small packages it is not a big deal since the download appears soon enough. However for large packages users end up clicking the button and have to wait several minutes for any indication that the download is actually going through. This is creating confusion for our users since it appears the downloads are not working.

Can this "background" download mechanism be modified through the config? Or is there some way to add an indicator that the download is actually happening?

Thanks!

**Comment by juanpicado** [comment]
Got it, is actually true maybe there is some component at material-ui we can use to provide better visual feedback

eg: https://mui.com/material-ui/react-progress/

like this

![Kapture 2023-10-14 at 11 03 07](https://github.com/verdaccio/verdaccio/assets/558752/30fa31db-302d-42fa-92e5-a06fee7300fd)

**Reply by jeremydiazmeta** [reply]
Would it also be possible to include a percentage somewhere?

**Reply by juanpicado** [reply]
I dunno how to achieve that with JS.

**Comment by juanpicado** [comment, marked as answer]
https://github.com/verdaccio/verdaccio/pull/4071
